src/secrets/CommandEdit.ts

```diff
@@ -54,41 +54,66 @@ class CommandEdit extends CommandPolykey {
         });
-        const response = await binUtils.retryAuthentication(
-          (auth) =>
-            pkClient.rpcClient.methods.vaultsSecretsGet({
-              metadata: auth,
-              nameOrId: secretPath[0],
-              secretName: secretPath[1],
-            }),
-          meta,
-        );
-        const secretContent = response.secretContent;
+        const [secretExists, secretContent] =
+          await binUtils.retryAuthentication(async (auth) => {
+            let secretExists: boolean = true;
+            let response;
+            try {
+              response = await pkClient.rpcClient.methods.vaultsSecretsGet({
+                metadata: auth,
+                nameOrId: secretPath[0],
+                secretName: secretPath[1],
+              });
+            } catch (e) {
+              if (e.cause.name === 'ErrorSecretsSecretUndefined') {
+                secretExists = false;
+              } else {
+                throw e;
+              }
+            }
+            return [secretExists, response?.secretContent];
+          }, meta);
         const tmpDir = await fs.promises.mkdtemp(
           path.join(os.tmpdir(), 'polykey-'),
         );
         const tmpFile = path.join(tmpDir, 'pksecret');
-        await this.fs.promises.writeFile(tmpFile, secretContent);
+        if (secretExists) {
+          await this.fs.promises.writeFile(tmpFile, secretContent);
+        }
         execSync(`$EDITOR \"${tmpFile}\"`, { stdio: 'inherit' });
-        let content: Buffer;
+        let content: string;
         try {
-          content = await this.fs.promises.readFile(tmpFile);
+          const tmpContent = await this.fs.promises.readFile(tmpFile);
+          content = tmpContent.toString('binary');
+          if (!secretExists) {
+            await binUtils.retryAuthentication(async (auth) => {
+              await pkClient.rpcClient.methods.vaultsSecretsNew({
+                metadata: auth,
+                nameOrId: secretPath[0],
+                secretName: secretPath[1],
+                secretContent: content,
+              });
+              await pkClient.rpcClient.methods.vaultsSecretsEdit({
+                metadata: auth,
+                nameOrId: secretPath[0],
+                secretName: secretPath[1],
+                secretContent: content,
+              });
+            });
+          }
         } catch (e) {
-          throw new errors.ErrorPolykeyCLIFileRead(e.message, {
-            data: {
-              errno: e.errno,
-              syscall: e.syscall,
-              code: e.code,
-              path: e.path,
-            },
-            cause: e,
-          });
+          if (secretExists) {
+            throw new errors.ErrorPolykeyCLIFileRead(e.message, {
+              data: {
+                errno: e.errno,
+                syscall: e.syscall,
+                code: e.code,
+                path: e.path,
+              },
+              cause: e,
+            });
+          }
```

Comment:
This error handling doesn't make sense. You always need to check the error you're getting and never assume its going to be the error you think it is.  So always check the error you're getting then you can decide what to do with it. In this case we wrap the error. If its not the error we expect then we MUST re-throw that error. Never ignore an error unless its explicitly intended logic and always comment as to why. But its extremely rare that we just ignore errors.

This is important since improper handling of errors makes it extremely hard to debug a problem. You've seen it yourself today with that `read 0` error.